falcon/util/structures.py

```diff
@@ -105,3 +105,19 @@ def copy(self):
 
     def __repr__(self):
         return '%s(%r)' % (self.__class__.__name__, dict(self.items()))
+
+
+class ETag(str):
+    """An entity-tag ``str``-like object with the weakness indicator.
+
+    Attributes:
+        is_weak (bool): The weakness indicator.
+
+    """
+    is_weak = False
```

Comment:
I don't we need to use the discussed class variable shadowing trick here, if we make sure we always set it. Unless I missed important motivation to do it this way?

Reply:
It's a public class with public method `to_header` which require this attribute. Since we don't override the class constructor (it worse for performance) and don't ensure that it always will be created through a builder function we must define it. I mean that some users can used it for creating entity tags and set them as a header. Otherwise we should make this class private.